src/Agent.Worker/TestResults/TestRunDataPublisherHelper.cs

```diff
@@ -0,0 +1,145 @@
+﻿using Microsoft.TeamFoundation.TestClient.PublishTestResults;
+using Microsoft.TeamFoundation.TestManagement.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Worker.TestResults.Utils;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using LegacyTestRunPublisher = Microsoft.VisualStudio.Services.Agent.Worker.LegacyTestResults.ITestRunPublisher;
+using ITestResultsServer = Microsoft.VisualStudio.Services.Agent.Worker.LegacyTestResults.ITestResultsServer;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    public class TestRunDataPublisherHelper
```

Comment:
lot of code is getting duplicated now, initially I thought it's just an API call...
Now Thinking should have moved the code to PTR library and used it here ? 

Now we are increasing the maintenance cost, as the duplicate code is getting bigger.

Reply:
Let's discuss over it on monday. Looks big change now.

Reply:
That's Correct Moving it to library makes sense and reduce the maintenance cost.

Reply:
Are you planning to make this change?

Reply:
Created a [Task](https://dev.azure.com/mseng/AzureDevOps/_workitems/edit/1709151) will pick this up once customer is unblocked.